lisa/sut_orchestrator/azure/features.py

```diff
@@ -1905,3 +1905,77 @@ def can_disable(cls) -> bool:
 
     def enabled(self) -> bool:
         return True
+
+
+@dataclass_json()
+@dataclass()
+class ArchitectureSettings(schema.FeatureSettings):
+    type: str = "Architecture"
+    # Architecture in hyper-v
+    arch: str = "x64"
+
+    def __eq__(self, o: object) -> bool:
+        if not super().__eq__(o):
+            return False
+
+        assert isinstance(o, ArchitectureSettings), f"actual: {type(o)}"
+        return self.type == o.type and self.arch == o.arch
+
+    def __repr__(self) -> str:
+        return f"arch:{self.arch}"
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __hash__(self) -> int:
+        return super().__hash__()
+
+    def _get_key(self) -> str:
+        return f"{super()._get_key()}/{self.arch}"
+
+    def check(self, capability: Any) -> search_space.ResultReason:
+        assert isinstance(
+            capability, ArchitectureSettings
+        ), f"actual: {type(capability)}"
+        result = super().check(capability)
+
+        result.merge(
+            search_space.check_setspace(self.arch, capability.arch),
+            "vhd Architecture",
+        )
+
+        return result
+
+    def _call_requirement_method(self, method_name: str, capability: Any) -> Any:
+        assert isinstance(
+            capability, ArchitectureSettings
+        ), f"actual: {type(capability)}"
+
+        value = ArchitectureSettings()
+        if self.arch or capability.arch:
+            value.arch = getattr(search_space, f"{method_name}_setspace_by_priority")(
+                self.arch, capability.arch, ["x64", "Arm64"]
```

Comment:
Do we need to handle AMD as well?

Reply:
Thanks for the point, I think `x86-64 (also known as x64, x86_64, AMD64, and Intel 64) `, but will double check the return value of amd64 size.

Reply:
I feel it would be worth separating intel and AMD. For example, I saw failures for nested cases in AMD but not intel. This would make it easy to see that certain failures are CPU vendor specific

Reply:
This field is for CPU arch only. The VHD with different arc maps to different VM sizes. The CPU info is a kind of information. We haven't collected it, but it's more related to vm sizes, not related to vhd.

Reply:
@somil55 only two values return from API level (image/vmsize), x64 and Arm64, for your case, we may think about how to achieve it in another way.

Reply:
@somil55 agree?